src/ir/auth_logic/ast.cc

```diff
@@ -0,0 +1,90 @@
+//----------------------------------------------------------------------------
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//----------------------------------------------------------------------------
+
+#include "src/ir/auth_logic/ast.h"
+
+namespace raksha::ir::auth_logic {
+
+// A potentially ugly print of the state in this class
+// for debugging only
+std::string BaseFact::DebugPrint() const {
```

Comment:
When we have the equality operators, will we need these debug print for tests at all?

Reply:
No the `DebugPrint` isn't needed for anything -- it will be technically dead code but it could be used for debugging. Would you rather I delete all the DebugPrints?